azurerm/internal/services/network/client.go

```diff
@@ -27,6 +27,7 @@ type Client struct {
 	SecurityGroupClient                  *network.SecurityGroupsClient
 	SecurityRuleClient                   *network.SecurityRulesClient
 	SubnetsClient                        *network.SubnetsClient
+	NatGatewaysClient                    *network.NatGatewaysClient
```

Comment:
Can we make this just `NatGatewayClient` instead of `NatGatewaysClient`?
```suggestion
	NatGatewayClient                    *network.NatGatewayClient
```

Reply:
In SDK, client often appears in plural form, and so does this.
For example, the subnetsClient above this line is also in plural form.

Reply:
Look at the two above that, `SecurityRuleClient` and `SecurityGroupClient`, which were plural in the SDK and changed to be singular in the provider, I think that the `SubnetsClient` was just missed on the original PR... that said, do you think we update this to `NatGatewayClient`?

Reply:
Changed